desktop/test/run-visdiff.js

```diff
@@ -0,0 +1,185 @@
+import process from 'process'
+import fs from 'fs'
+import path from 'path'
+import {execSync} from 'child_process'
+import _ from 'lodash'
+import gm from 'gm'
+import github from 'octonode'
+
+const BUCKET_S3 = 's3://keybase-app-visdiff'
+const BUCKET_HTTP = 'http://keybase-app-visdiff.s3.amazonaws.com'
+const MAX_INLINE_IMAGES = 6
+
+const DIFF_NEW = 'new'
+const DIFF_REMOVED = 'removed'
+const DIFF_CHANGED = 'changed'
+const DIFF_SAME = 'same'
+
+function renderScreenshots (commitRange) {
+  for (const commit of commitRange) {
+    console.log(`Rendering screenshots of ${commit}`)
+    execSync(`git checkout -f ${commit} && mkdir -p screenshots/${commit} && npm run render-screenshots -- screenshots/${commit}`)
+  }
+  execSync(`git checkout -f ${commitRange[1]}`)
+}
+
+function compareScreenshots (commitRange, diffDir, callback) {
+  const results = {}
+
+  execSync(`mkdir -p screenshots/${diffDir}`)
+
+  const files0 = fs.readdirSync(`screenshots/${commitRange[0]}`)
+  const files1 = fs.readdirSync(`screenshots/${commitRange[1]}`)
+  const files = _.union(files0, files1)
+  function compareNext () {
+    const filename = files.pop()
+    if (!filename) {
+      callback(results)
+      return
+    }
+
+    if (filename.startsWith('.')) {
+      return
+    }
+
+    const diffPath = `screenshots/${diffDir}/${filename}`
+
+    const oldPath = `screenshots/${commitRange[0]}/${filename}`
+    if (!fs.existsSync(oldPath)) {
+      results[diffPath] = DIFF_NEW
+      compareNext()
+      return
+    }
+
+    const newPath = `screenshots/${commitRange[1]}/${filename}`
+    if (!fs.existsSync(newPath)) {
+      results[diffPath] = DIFF_REMOVED
+      compareNext()
+      return
+    }
+
+    const compareOptions = {
+      tolerance: 1e-6,  // leave a little wiggle room for antialiasing inconsistencies
+      file: diffPath
+    }
+    gm.compare(oldPath, newPath, compareOptions, (err, isEqual) => {
+      if (err) {
+        console.log(err)
+        process.exit(1)
+      }
+      results[diffPath] = isEqual ? DIFF_SAME : DIFF_CHANGED
+      compareNext()
+    })
+  }
+  compareNext()
```

Comment:
we have some use cases of promises but only saga-related generators so i'd avoid those if possible